### PR TITLE
Exit early when partitions are empty during export data

### LINKED ISSUE #668: Export data hangs when empty partitions exist on Oracle/MySQL
yb-voyager does not exit during export data when a partition is empty. We need to explicitly exit in these cases.

### PR DESCRIPTION
**Pull request by rahulb-yb**
[Fixes #668]
During export data, if a partition is empty (contains no data within it), yb-voyager would hang, expecting a file corresponding to this partition to be exported. ora2pg no longer exports data files for empty partitions, owing to the need for this patch to be brought up.
This patch adds a new channel which allows us to know when `ora2pg` has successfully exit, making use of which, we can allow `yb-voyager` to safely exit.
Automation tests for the same to be covered as a part of #451

### CODE REVIEW

File: yb-voyager/src/srcdb/postgres.go [hunk 1]
```diff
@@ -175,8 +175,8 @@ func (pg *PostgreSQL) ExportSchema(exportDir string) {
 	pgdumpExtractSchema(pg.source, pg.getConnectionUri(), exportDir)
 }
 
-func (pg *PostgreSQL) ExportData(ctx context.Context, exportDir string, tableList []string, quitChan chan bool, exportDataStart chan bool) {
-	pgdumpExportDataOffline(ctx, pg.source, pg.getConnectionUri(), exportDir, tableList, quitChan, exportDataStart)
+func (pg *PostgreSQL) ExportData(ctx context.Context, exportDir string, tableList []string, quitChan chan bool, exportDataStart, exportSuccessChan chan bool) {
+	pgdumpExportDataOffline(ctx, pg.source, pg.getConnectionUri(), exportDir, tableList, quitChan, exportDataStart) // No need to use exportSuccessChan
```
Comment (amit-yb) [inline yb-voyager/src/srcdb/postgres.go:179]: Let's have a uniform behaviour: `exportSuccessChan` is signalled whenever the export data tool (ora2pg/pg_dump) completes successfully.
That should allow us getting rid of explicit checks for the Oracle/MySQL DBType.

File: yb-voyager/cmd/exportDataStatus.go [hunk 2]
```diff
@@ -152,13 +159,20 @@ func exportDataStatus(ctx context.Context, tablesProgressMetadata map[string]*ut
 			fmt.Println(ctx.Err())
 			break
 		}
+		// Wait for metadata to update and then exit (fix for empty partitions in MySQL, Oracle)
+		if updateMetadataAndExit && (source.DBType == MYSQL || source.DBType == ORACLE) {
```
Comment (amit-yb) [inline yb-voyager/cmd/exportDataStatus.go:163]: Get rid of the explicit check on the DBType. Have a consistent behaviour irrespective of source DB Type.

File: yb-voyager/cmd/exportData.go [hunk 3]
```diff
@@ -138,14 +139,17 @@ func exportDataOffline() bool {
...
 	<-exportDataStart
 
 	updateFilePaths(&source, exportDir, tablesProgressMetadata)
 	if !disablePb {
 		utils.WaitGroup.Add(1)
-		exportDataStatus(ctx, tablesProgressMetadata, quitChan)
+		exportDataStatus(ctx, tablesProgressMetadata, quitChan, exportSuccessChan)
+	} else if source.DBType == MYSQL || source.DBType == ORACLE {
+		// TODO: Update tables metadata when PB are disabled.
+		<-exportSuccessChan
```
Comment (amit-yb) [inline yb-voyager/cmd/exportData.go:152]: Why is this required?

Comment (rahulb-yb) [reply yb-voyager/cmd/exportData.go:152]: Will change this notification channel to a buffered channel, we will need to change some other notification channels to buffered channels as well in a separate PR; we can probably accommodate this change in the PR for the interface-based implementation for the progress bars

Comment (amit-yb) [reply yb-voyager/cmd/exportData.go]: I would suggest to get this change, at least, for the `exportSuccessChan`. It will be very difficult to keep track of it once this PR gets merged.

**Review by amit-yb — APPROVED** [review]
LGTM.

File: yb-voyager/cmd/exportData.go (hunk 3 above)
Comment (amit-yb) [reply yb-voyager/cmd/exportData.go]: Ignore the previous comment--you have already taken care of it. Approved the PR.
